src/fee_lab_to_nwb/happ_ecephys/happmotifinterface.py

```diff
@@ -0,0 +1,62 @@
+from typing import Optional
+
+import numpy as np
+from neuroconv.basedatainterface import BaseDataInterface
+from pynwb import NWBFile
+from scipy.io import loadmat
+
+
+class MotifInterface(BaseDataInterface):
+    """Data interface for adding timing of the motifs as trials to the NWB file."""
+
+    def __init__(self, file_path: str):
+        """
+        Create the interface for writing the timing of the motifs to the NWB file.
+        The motifs are added as trials.
+
+        Parameters
+        ----------
+        file_path: str
+            The path to the file containing the timing of the motifs.
+        """
+        super().__init__(file_path=file_path)
+
+    def read_motif_data(self):
+        """Reads the .mat file containing the timing of the motifs.
+        Returns the identifier and timing of the motifs."""
+        motif_data = loadmat(self.source_data["file_path"])
+        assert "motifTimingData" in motif_data, "'motifTimingData' should be in file."
+
+        motifs = motif_data["motifTimingData"]
+
+        num_motifs = motifs.shape[0]
+        motif_ids = [motifs[:, 0][motif_num][0] for motif_num in range(num_motifs)]
+        motif_timestamp = [
+            motifs[:, 1][motif_num][0][0] for motif_num in range(num_motifs)
+        ]
+        return motif_ids, motif_timestamp
+
+    def run_conversion(
+        self,
+        nwbfile: Optional[NWBFile] = None,
```

Comment:
It looks like you are expecting `nwbfile` to be an `NWBFile` here, so the `Optional` annotation is not correct

Reply:
indeed, thank you!